src/group_theory/index.lean

```diff
@@ -161,6 +161,54 @@ begin
   exact ⟨fintype.card H, H.index_mul_card.symm⟩,
 end
 
+lemma relindex_eq_zero_of_le  (h : K ≤ L) (h2 : H.relindex K = 0) : H.relindex L = 0 :=
+cardinal.to_nat_eq_zero_of_injective (quotient_group.le_quot_map H h ).injective h2
+
+lemma index_eq_zero_of_le {H K : subgroup G} (h : H ≤ K) (h1 : K.index = 0) : H.index = 0 :=
+by rw [←subgroup.relindex_mul_index h, h1, mul_zero]
+
+lemma inf_relindex_inf : (H ⊓ K).relindex (K ⊓ L) = H.relindex (K ⊓ L) :=
```

Comment:
The lemmas `inf_relindex_inf`, `inf_relindex_subgroup_of`, and `inf_ind_prod` are no longer used in this PR, so you might consider splitting them off into a separate PR.